app/processing/transform-payment-request.js

```diff
@@ -14,17 +14,17 @@ const transformPaymentRequest = async (paymentRequest) => {
   // Check to see if payment request has had a previous payment request.
   // if yes, need to treat as post payment adjustment and calculate Delta which can result in payment request splitting
   const previousPaymentRequests = await getCompletedPaymentRequests(paymentRequest)
+  const sanitizedPaymentRequest = applyDualAccounting(paymentRequest, previousPaymentRequests)
```

Comment:
Moving dual accounting to before delta calculation.

Reason is that delta is performed at line level which includes grouping by fund code.

If we leave after then we'd have current payment request as `EGF00` and previous all with `DOM10` (for example).  This means we'd always end up creating a negative invoice line for the domestic fund to repay under EU fund code.  Then when dual account is applied we'd end up changing both lines to domestic.  The end result would be fine, but it makes the payment request data more difficult to work with.

Example if we left it after post payment adjustments.

Current payment request -> `EGF00 -> £100`
Previous payment request -> `DOM10` -> £500`

Delta calculation grouped by fund would result in
`EGF00 -> £100`
`DOM10 -> -£500`

After dual accounting would be:
`DOM10 -> £100`
`DOM10 -> -£500`

The result is fine as the customer gets a reduction of -£400 against a domestic fund.  But we've now split that over two lines for no reason.

If however, we apply the domestic fund before delta.  We'd do this.

Current payment request -> `EGF00 -> £100`
Previous payment request -> `DOM10` -> £500`

Apply dual accounting:
Current payment request -> `DOM10 -> £100`
Previous payment request -> `DOM10 -> £500`

Then delta calculation would output
`DOM10 -> -£400`

Same overall result, but much cleaner to read.

Reply:
I agree, it is much cleaner to read :)